src/Options.ts

```diff
@@ -30,10 +30,10 @@ interface IProcess {
 declare var process: IProcess;
 
 let OptionsVals: IOptions = {
-    setTimeout: setTimeout.bind(null),
-    clearTimeout: clearTimeout.bind(null),
+    setTimeout: (...args: any[]) => setTimeout(...args),
```

Comment:
I think you don't need to define types for `setTimeout` and `clearTimeout` and because `OptionsVals` refers to `IOptions`

```typescript
let OptionsVals: IOptions = {
    setTimeout: (callback, timeout) => setTimeout(callback, timeout),
    clearTimeout: id => clearTimeout(id),
    shouldComponentUpdateComparator: isEqual,

    preventTryCatchInRender: false,
    development: typeof process !== 'undefined' && process.env && process.env.NODE_ENV !== 'production'
};
```